File: src/template/cpp/dfs/dfs.cpp
#include <vector>
using namespace std;

vector<int> dfs(int start, const vector<vector<int>>& adj) {
    int n = adj.size();
    vector<bool> vis(n);
    vector<int> order, st;
    st.reserve(n);
    st.push_back(start);
    while (!st.empty()) {
        int u = st.back();
        st.pop_back();
        if (vis[u]) continue;
        vis[u] = true;
        order.push_back(u);
        for (int i = (int)adj[u].size() - 1; i >= 0; --i) {
            int v = adj[u][i];
            if (!vis[v]) st.push_back(v);
        }
    }
    return order;
}

File: src/template/julia/dfs/dfs.jl
function dfs(start::Int, adj::Vector{Vector{Int}})
    n = length(adj)
    vis = falses(n)
    order = Int[]
    st = [start]
    while !isempty(st)
        u = pop!(st)
        if vis[u]
            continue
        end
        vis[u] = true
        push!(order, u)
        for v in reverse(adj[u])
            if !vis[v]
                push!(st, v)
            end
        end
    end
    return order
end